pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -306,38 +368,66 @@ func determineCandidatesWithMinimalDistanceStrategy(seeds []*gardencorev1beta1.S
 	}
 
 	var (
-		maxMatchingCharacters int
-		shootRegion           = shoot.Spec.Region
-		candidates            []*gardencorev1beta1.Seed
+		minDistance = 1000
+		shootRegion = shoot.Spec.Region
+		candidates  []*gardencorev1beta1.Seed
 	)
 
 	for _, seed := range seeds {
-		if seed.DeletionTimestamp == nil && seed.Spec.Provider.Type == shoot.Spec.Provider.Type && !gardencorev1beta1helper.TaintsHave(seed.Spec.Taints, gardencorev1beta1.SeedTaintInvisible) && common.VerifySeedReadiness(seed) {
-			seedRegion := seed.Spec.Provider.Region
-			var matchingCharacters int
-			for i := 0; i <= len(shootRegion); i++ {
-				prefix := shootRegion[:i]
-				if strings.HasPrefix(seedRegion, prefix) {
-					matchingCharacters = len(prefix)
-					continue
-				}
-			}
+		seedRegion := seed.Spec.Provider.Region
+		dist, _ := distance(seedRegion, shootRegion)
 
-			// append
-			if matchingCharacters == maxMatchingCharacters {
-				candidates = append(candidates, seed)
-				continue
-			}
-			// replace
-			if matchingCharacters > maxMatchingCharacters {
-				maxMatchingCharacters = matchingCharacters
-				candidates = []*gardencorev1beta1.Seed{seed}
-			}
+		// append
+		if dist == minDistance {
+			candidates = append(candidates, seed)
+			continue
+		}
+		// replace
+		if dist < minDistance {
+			minDistance = dist
+			candidates = []*gardencorev1beta1.Seed{seed}
 		}
 	}
 	return candidates
 }
 
+var orientations = []string{"north", "south", "east", "west", "central"}
+
+// distance extracts an orientation relative to a base from a region name
+func orientation(name string) (normalized string, orientation string) {
+	for _, o := range orientations {
+		if i := strings.Index(name, o); i >= 0 {
+			orientation = o
+			normalized = name[:i] + ":" + name[i+len(o):]
+			return
+		}
+	}
+	return name, ""
+}
+
+// distance calculates a formal distance between two region names observing
+// some usual orientation keywords. It is based on the levenshtein distance
+// of the regions base names plus the difference based on the orientation.
+// regions with the same base but different orientations are basically nearer
+// to each other than two completely diffrent regions.
+func distance(seed, shoot string) (int, int) {
```

Comment:
Can we get rid of second return type? I think it makes the signature more complex and it's not really used except by some unit tests.

Reply:
for testing the internal values are required, but I introduced an `end user`function that has only one return value